### PR TITLE
Post spec

### PR DESCRIPTION
**Pull request by rachjgriff**
Created a unit test for formatted_date_time method in Post Model
Updated posts/index.html.erb to include formatted_date_time method

Rubocop updates (NO OFFENCES!!!):
Updated user_email method in Post Model to `delegate :email, :to => :user, :allow_nil => true`
Updated User model to incl. dependent: :destroy - this means that if a user is deleted, all associated posts are deleted.

Merged in master

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,9 +1,7 @@
 class Post < ApplicationRecord
   belongs_to :user
 
-  def user_email
-    user.email
-  end
+  delegate :email, :to => :user, :allow_nil => true
```
Comment (Danielandro) [inline app/models/post.rb]: I'm assuming this does the same thing as your `user_email` method? How, what & who?

File: app/models/user.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable, :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
-  has_many :posts
-  
+
+  has_many :posts, dependent: :destroy
```
Comment (Danielandro) [inline app/models/user.rb]: 😄

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 <% @posts.each do |post| %>
   <p><%= simple_format post.message %></p>
-  <%= post.created_at.strftime("%d %B %Y %H:%M") %>
+  <%= post.formatted_date_time %>
```
Comment (Danielandro) [inline app/views/posts/index.html.erb]: Its so beautiful 😍

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -1,5 +1,14 @@
 require 'rails_helper'
```
Comment (Danielandro) [inline spec/models/post_spec.rb]: I really like this test - super cool

File: app/models/post.rb (hunk 1 above)
Comment (zepherine2006DEV) [reply app/models/post.rb]: Very clever!

**Pull request merged**